substrate-node/pallets/pallet-smart-contract/src/cost.rs

```diff
@@ -193,6 +209,22 @@ pub fn calculate_resources_cost<T: Config>(
 
     return total_cost.ceil().to_num::<u64>();
 }
+
+// Calculates the cost of extra fee for a dedicated node.
+// TODO: check why this gives different results when giving it a low input (multiple billing cycles) and a high input (total seconds elapsed)
```

Comment:
Actually yes, so we still need to check what's the problem here.

Reply:
Is there an issue for it? :)

Reply:
I guess it's truncating issue when we deal with doubles
Since we do a double division which is converted to `u64` each time we call this function we get rid of some small "after comma" quantity.
So for the same elapsed time, calling once or calling multiple times and aggregating, will not give the same results.
To limit this error (which is one of the reason we use usd units instead of musd I guess) I also suggest to use a `round()` before converting to `u64`

Reply:
see https://github.com/threefoldtech/tfchain/issues/736